src/lib/upgradePackageDefinitions.ts

```diff
@@ -11,6 +11,77 @@ import { pickBy } from './pick'
 import queryVersions from './queryVersions'
 import upgradeDependencies from './upgradeDependencies'
 
+/**
+ * check if in peer violation
+ *
+ * @returns
+ */
+const checkIfInPeerViolation = (
+  currentDependencies: Index<VersionSpec>,
+  filteredUpgradedDependencies: Index<VersionSpec>,
+  upgradedPeerDependencies: Index<Index<VersionSpec>>,
+) => {
+  const upgradedDependencies = { ...currentDependencies, ...filteredUpgradedDependencies }
+  const upgradedDependenciesVersions = Object.fromEntries(
+    Object.entries(upgradedDependencies).map(([packageName, versionSpec]) => {
+      return [packageName, semver.minVersion(versionSpec)?.version ?? versionSpec]
+    }),
+  )
+  const filteredUpgradedPeerDependencies = { ...upgradedPeerDependencies }
+  let wereUpgradedDependenceFiltered = false
+  const filteredUpgradedDependenciesAfterPeers = pickBy(filteredUpgradedDependencies, (spec, dep) => {
+    const peerDeps = filteredUpgradedPeerDependencies[dep]
+    if (!peerDeps) {
+      return true
+    }
+    const valid = Object.entries(peerDeps).every(
+      ([peer, peerSpec]) =>
+        upgradedDependenciesVersions[peer] === undefined ||
+        semver.satisfies(upgradedDependenciesVersions[peer], peerSpec),
+    )
+    if (!valid) {
+      wereUpgradedDependenceFiltered = true
+      delete filteredUpgradedPeerDependencies[dep]
+    }
+    return valid
+  })
+  return {
+    issuesFound: wereUpgradedDependenceFiltered,
+    filteredUpgradedDependencies: filteredUpgradedDependenciesAfterPeers,
+    upgradedPeerDependencies: filteredUpgradedPeerDependencies,
+  }
+}
+
+/**
+ * rerun upgrade if changed peers
+ *
+ * @returns
+ */
+const rerunUpgradeIfChangedPeers = async (
+  currentDependencies: Index<VersionSpec>,
+  filteredUpgradedDependencies: Index<VersionSpec>,
+  upgradedPeerDependencies: Index<Index<VersionSpec>>,
+  latestVersionResults: Index<VersionResult>,
+  options: Options,
+): Promise<void | [Index<VersionSpec>, Index<VersionResult>, Index<Index<VersionSpec>>?]> => {
+  if (Object.keys(filteredUpgradedDependencies).length === 0) {
+    return [{}, latestVersionResults, options.peerDependencies]
+  }
+  const peerDependencies = { ...options.peerDependencies, ...upgradedPeerDependencies }
+  if (!dequal(options.peerDependencies, peerDependencies)) {
```

Comment:
This condition would be better inverted:

```ts
if (dequal(options.peerDependencies, peerDependencies)) return null

...
```

Reply:
agree, done